packages/cli/src/commands/validator/missed-blocks.ts

```diff
@@ -0,0 +1,149 @@
+import { Address } from '@celo/contractkit'
+import { concurrentMap } from '@celo/utils/lib/async'
+import { bitIsSet, parseBlockExtraData } from '@celo/utils/lib/istanbul'
+import { flags } from '@oclif/command'
+import { cli } from 'cli-ux'
+import { readFileSync } from 'fs'
+import { BaseCommand } from '../../base'
+import { Flags } from '../../utils/command'
+import { ElectionResultsCache } from '../../utils/election'
+
+interface VerboseValidatorEntry {
+  name: string
+  group: string
+  address: Address
+  missedBlocks: number[]
+}
+
+interface ValidatorEntry {
+  address: Address
+  missedBlocks: number[]
+}
+
+export const verboseStatusTable = {
+  address: {},
+  name: {},
+  group: {},
+  missedBlocks: {},
+}
+
+export const statusTable = {
+  address: {},
+  missedBlocks: {},
+}
+
+export default class ValidatorSignedBlocks extends BaseCommand {
+  static description =
+    'Display a list of blocks that each validator missed. Can use the genesis to not rely on contracts, but contracts works better.'
+
+  static flags = {
+    ...BaseCommand.flagsWithoutLocalAddresses(),
+    genesis: Flags.path({
+      description:
+        'path to the genesis block to get the initial validator set, uses contracts if not provided.',
+    }),
+    'at-block': flags.integer({
+      description: 'latest block to examine for signer activity',
+    }),
+    lookback: flags.integer({
+      description: 'how many blocks to look back for signer activity',
+      default: 120,
+    }),
+  }
+
+  static examples = [
+    'downtime --genesis ./genesis.json',
+    'downtime',
+    'downtime --at-block 100000 --genesis ./genesis.json',
+    'downtime --lookback 500 --genesis ./genesis.json',
+  ]
+
+  async run() {
+    const res = this.parse(ValidatorSignedBlocks)
+
+    const latest = res.flags['at-block']
+      ? res.flags['at-block'] + 1
+      : (await this.web3.eth.getBlock('latest')).number
+
+    let electionCache
+    let validators, accounts
+    let haveContracts
+    let electedSigners: string[]
+    const validatorToGroupName = new Map<Address, string>()
+    // Try to get contracts, if not try to fall back to genesis block
+    try {
+      const election = await this.kit.contracts.getElection()
+      accounts = await this.kit.contracts.getAccounts()
+      validators = await this.kit.contracts.getValidators()
+      const vgroups = await validators.getRegisteredValidatorGroups()
```

Comment:
No entirely sure. But i guess one can 'deregister'  groups. Meaning that you need to get the registered group "as were registered in that point in time". Specially if you go deep back in time

Reply:
Should be fixed, had to make some minor modifications to contract kit